pp.c

```diff
@@ -5380,6 +5380,31 @@ PP(pp_exists)
     RETPUSHNO;
 }
 
```

Comment:
Some documentation above this about how to use it and its limitations would be nice. It seems to me that this won't handle magic in an expected way. The commit says it's roughly equivalent to `exists .... ? ... : ... `but I don't think EXISTS magic would ever be called since the code is doing just a fetch or a delete.

Reply:
Oooh, fun. I hadn't thought about magic (well, overload) since in my use-case I know that the HV is a plain one because it's internal to the surrounding code. But yes, perhaps an overload check for that and calling something else on it would be useful indeed.

Reply:
Looks like you fixed this up. Nice. It would be cool to document how/when various magic would be called so it's not a surprise, but I think this is fine as is if you don't want to.

I think it's something like:

If the hash is magical, this will invoke EXISTS on it. If the entry is found and the OPpHELEMEXISTSOR_DELETE flag is true, DELETE will be invoked. Otherwise FETCH will be.

Reply:
Added extra paragraph about tied hashes.

Reply:
Perfect, thanks